usaspending_api/awards/models.py

```diff
@@ -334,7 +334,7 @@ class TransactionContract(DataSourceTrackedModel):
     naics = models.TextField(blank=True, null=True, verbose_name="NAICS", help_text="Specified which industry the work for this transaction falls into. A 6-digit code")
     naics_description = models.TextField(blank=True, null=True, verbose_name="NAICS Description", help_text="A plain text description of the NAICS code")
     period_of_performance_potential_end_date = models.DateField(max_length=10, verbose_name="Period of Performance Potential End Date", null=True, help_text="The potential end date of the period of performance")
-    ordering_period_end_date = models.CharField(max_length=8, blank=True, null=True, help_text="The end date for the ordering period")
+    ordering_period_end_date = models.TextField(blank=True, null=True, help_text="The end date for the ordering period")  # If we aren't converting it to a date, do we even really want it?
```

Comment:
Well, there's nothing that actually makes `CAST(ordering_period_end_date as date)` fail in the production broker DB right now, so I can go ahead with that.  There are some odd historical-fiction or sci-fi dates.  What do we do with data quality issues?

```
%%sql $DATA_BROKER_DATABASE_URL
      select ordering_period_end_date, 
      cast(ordering_period_end_date as date) AS dt,
      count(*)
from  award_procurement 
where ordering_period_end_date is not null
and   (   CAST(ordering_period_end_date as date) < '2000-01-01'
       or CAST(ordering_period_end_date as date) > '2030-01-01')
group by 1, 2
order by 2

ordering_period_end_date	dt	count
10170106	1017-01-06	4
10171029	1017-10-29	25
19000101	1900-01-01	10424
20300530	2030-05-30	56
20301001	2030-10-01	1
20301130	2030-11-30	10
20301230	2030-12-30	16
20310331	2031-03-31	40
20370614	2037-06-14	36
20411029	2041-10-29	12
20420326	2042-03-26	1
20611221	2061-12-21	1
21060127	2106-01-27	1
21060929	2106-09-29	2
21061020	2106-10-20	1
21070330	2107-03-30	3
21070929	2107-09-29	5
29170530	2917-05-30	2
30160929	3016-09-29	20
30170429	3017-04-29	1
30180330	3018-03-30	6
```